plugins/modules/proxmox_backup_schedule.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2025 Marzieh Raoufnezhad <raoufnezhad@gmail.com>
+# Copyright (c) 2024 Maryam Mayabi <mayabi.ahm at gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: proxmox_backup_schedule
+
+short_description: Scheduling VM Backups and Removing it.
+
+version_added: 10.3.0
+
+description: The proxmox_backup_schedule module modify backup jobs such as set or delete vmid.
+
+author:
+  - "Marzieh Raoufnezhad (@raoufnezhad) <raoufnezhad@gmail.com>"
+  - "Maryam Mayabi (@mmayabi) <mayabi.ahm@gmail.com>"
+
+options:
+  vm_name:
+    description:
+      - The name of the Proxmox VM.
+      - Mutually exclusive with O(vm_id).
+    required: false
+    type: str
+  vm_id:
+    description:
+      - The ID of the Proxmox VM.
+      - Mutually exclusive with O(vm_name).
+    required: false
+    type: str
+  backup_id:
+    description: The backup job ID.
+    required: false
+    type: str
+  backup_action:
+    description:
+        - If V(update_vmid), the module will update backup job with new VM ID.
+        - If V(delete_vmid), the module will remove the VM ID from all backup jobs where the VM ID has existed.
+    required: true
+    type: str
+
+extends_documentation_fragment:
+  - community.general.proxmox.documentation
+  - community.general.attributes
+  - community.general.attributes.info_module
+  - community.general.proxmox.actiongroup_proxmox
+"""
+
+EXAMPLES = """
+- name: Scheduling VM Backups based on VM name.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_name: 'VM Name'
+    backup_id: 'backup-b2adffdc-316e'
+    backup_action: 'update_vmid'
+
+- name: Scheduling VM Backups based on VM ID.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_id: 'VM ID'
+    backup_id: 'backup-b2adffdc-316e'
+    backup_action: 'update_vmid'
+
+- name: Removing backup setting based on VM name.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_name: 'VM Name'
+    backup_action: 'delete_vmid'
+
+- name: Removing backup setting based on VM ID.
+  proxmox_backup_schedule:
+    api_user: 'myUser@pam'
+    api_password: '*******'
+    api_host: '192.168.20.20'
+    vm_id: 'VM ID'
+    backup_action: 'delete_vmid'
+"""
+
+RETURN = """
+---
+backup_schedule:
+  description:
+    - If V(update_vmid), the backup_schedule will return True after adding the VM ID to the backup job.
+    - If V(delete_vmid), the backup_schedule will return a list of backup job IDs where the VM ID has existed after removing it.
+  returned: always, but can be empty
+  type: dict
```

Comment:
Firstly, not a good idea to have a RV that results in different types - trust me, I have done that, and still needs fixing (module xfconf).
Secondly, you are declaring the type as `dict`, which neither of the returned values are.

I would suggest you make two different RVs, one for the update case, another for the delete case, then make them of the right type.